src/crafting.cpp

```diff
@@ -923,17 +922,12 @@ double player::crafting_success_roll( const recipe &making ) const
 
     // It's tough to craft with paws.  Fortunately it's just a matter of grip and fine-motor,
     // not inability to see what you're doing
-    if( has_trait( trait_PAWS ) || has_trait( trait_PAWS_LARGE ) ) {
-        int paws_rank_penalty = 0;
-        if( has_trait( trait_PAWS_LARGE ) ) {
-            paws_rank_penalty += 1;
-        }
-        if( making.skill_used == skill_id( "electronics" )
-            || making.skill_used == skill_id( "tailor" )
-            || making.skill_used == skill_id( "mechanics" ) ) {
-            paws_rank_penalty += 1;
+    for( const std::pair< trait_id, trait_data > &mut : my_mutations ) {
+        for( const std::pair<skill_id, int> &skib : mut.first->craft_skill_bonus ) {
+            if( making.skill_used == skib.first ) {
```

Comment:
`craft_skill_bonus` is a `std::map`. You can lookup keys in a map, there is a function for that. You do not need to iterate over all entries of the map.